tests/test_minimal/test_metadata_schema.py

```diff
@@ -0,0 +1,285 @@
+                ),
+            ),
+            SegmentationImages=dict(
+                name="SegmentationImages",
+                PlaneSegmentationChan1Plane0=dict(
+                    raw=dict(name="PlaneSegmentationChan1Plane0", description="raw segmentation image"),
+                    neuropil=dict(name="PlaneSegmentationChan1Plane0", description="neuropil segmentation image"),
+                ),
+            ),
+        ),
+    )
+
+    validate_metadata(metadata=metadata, schema=metadata_schema)
+
+
+invalid_plane_name = "plane_segmentation_chan1_plane0"
+name_error = "'name' is a required property"
+description_error = "'description' is a required property"
+plane_name_error_base = "does not match any of the regexes"
+not_enough_properties_error = "does not have enough properties"
+
+
+def test_invalid_ophys_metadata():
+    metadata_schema = load_dict_from_file(
+        Path(__file__).parent.parent.parent / "src" / "neuroconv" / "schemas" / "metadata_schema.json"
+    )
+
+    metadata = dict(
+        NWBFile=dict(
+            session_start_time="2020-01-01T00:00:00",
+            session_description="Auto-generated by neuroconv",
+            identifier="1234",
+        ),
+        Ophys=dict(
+            Devices=[],
+            Fluorescence={
+                "name": "Fluorescence",
+                "fluorescence_chan1_plane0": dict(),  # Value Unchecked
+                "FluorescenceChan1Plane0": dict(),
+                "FluorescenceChan1Plane1": dict(
+                    raw=dict(),
+                ),
+            },
+            DFOverF={
+                "name": "DfOverF",
+                "df_chan1_plane0": dict(),  # Value Unchecked
+                "DFChan1Plane0": dict(),
+                "DFChan1Plane1": dict(
+                    raw=dict(),
+                ),
+            },
+            SegmentationImages={
+                "name": "SegmentationImages",
+                "segmentation_chan1_plane0": dict(),  # Value Unchecked
+                "SegmentationChan1Plane0": dict(),
+                "SegmentationChan1Plane1": dict(
+                    raw=dict(),
+                ),
+            },
+        ),
+    )
+
+    validator = jsonschema.Draft7Validator(metadata_schema)
+
+    errors = [{"message": error.message, "path": error.json_path} for error in validator.iter_errors(metadata)]
+
+    def count_matching_elements(arr, condition_func):
+        filtered_elements = filter(condition_func, arr)
+        return len(list(filtered_elements))
+
+    nExpectedErrors = 11
+    assert len(errors) == nExpectedErrors
+
+    # One for each of Fluorescence, DFOverF, and SegmentationImages
+    assert (
+        count_matching_elements(
+            errors,
+            lambda o: f"'fluorescence_chan1_plane0' {plane_name_error_base}" in o["message"]
+            and o["path"] == "$.Ophys.Fluorescence",
+        )
+        == 1
+    )
```

Comment:
Still relying on a lot of implicit machinery here

I'd recommend the following pattern

```python
expected_errors = [
   {
        "message": "<insert full message, hard-coded>",
        "json_path": "<full, readable path for that message>"
    },
    ... # repeat for all messages you expect
]

for expected_error in expected_errors:
    assert expected_error in errors
```

That, combined with the assertion on the number of errors, is effectively testing equality of the lists but in a way more easily readable at a glance by anyone coming back to this months/years later

Reply:
Alright pushing now!